src/app/parcels/fetchParcelTableData.ts

```diff
@@ -30,52 +35,158 @@ export const getCongestionChargeDetailsForParcels = async (
+    }
+
+    const { data, error } = await query;
 
     if (error) {
+        console.error(error);
         throw new DatabaseError("fetch", "parcel table data");
     }
+    return data ?? [];
+};
+
+export const getParcelsData = async (
+    supabase: Supabase,
+    filters: Filter<ParcelsTableRow, any[]>[],
+    sortState: SortState<ParcelsTableRow>,
+    start?: number,
+    end?: number,
+    parcelIds?: string[]
+): Promise<ParcelsTableRow[]> => {
+    const processingData = await getParcelProcessingData(
+        supabase,
+        filters,
+        sortState,
+        start,
+        end,
+        parcelIds
+    );
+    const congestionCharge = await getCongestionChargeDetailsForParcels(processingData, supabase);
+    const formattedData = processingDataToParcelsTableData(processingData, congestionCharge);
+
+    return formattedData;
+};
+
+export const getParcelsCount = async (
+    supabase: Supabase,
+    filters: Filter<ParcelsTableRow, any>[]
+): Promise<number> => {
+    let query = supabase.from("parcels_plus").select("*", { count: "exact", head: true });
+
+    filters.forEach((filter) => {
+        if (filter.methodConfig.methodType === PaginationType.Server) {
+            query = filter.methodConfig.method(query, filter.state);
+        }
+    });
+
+    const { count, error } = await query;
+    if (error || count === null) {
+        console.error(error);
+        throw new DatabaseError("fetch", "parcels");
+    }
+    return count;
+};
+
+export const getParcelIds = async (
+    supabase: Supabase,
+    filters: Filter<ParcelsTableRow, any>[],
+    sortState: SortState<ParcelsTableRow>
+): Promise<string[]> => {
+    let query = supabase.from("parcels_plus").select("*");
+    filters.forEach((filter) => {
+        if (filter.methodConfig.methodType === PaginationType.Server) {
+            query = filter.methodConfig.method(query, filter.state);
+        }
+    });
+    if (sortState.sort && sortState.column.sortMethodConfig?.methodType === PaginationType.Server) {
+        query = sortState.column.sortMethodConfig.method(query, sortState.sortDirection);
+    } else {
+        query = query.order("packing_datetime", { ascending: false });
+    }
+    const { data, error } = await query;
+    if (error) {
+        console.error(error);
+        throw new DatabaseError("fetch", "parcels");
+    }
+    return data.map((parcel) => parcel.parcel_id ?? "") ?? [];
+};
 
+export const getAllValuesForKeys = async <Return>(
+    supabase: Supabase,
+    table: TableNames | ViewNames,
+    selectMethod: (
+        query: PostgrestQueryBuilder<Database["public"], any, any>
+    ) => PostgrestFilterBuilder<Database["public"], any, any>
+): Promise<Return> => {
+    const query = selectMethod(supabase.from(table));
+    const { data, error } = await query;
+    if (error) {
+        console.error(error);
+        throw new DatabaseError("fetch", table);
+    }
     return data ?? [];
 };
```

Comment:
I'm not sure I understand what this function is trying to achieve - the caller needs to define `selectMethod` anyway so is it easier to read if you just did this without a function? Maybe it's a naming problem? `getAllValuesForKeys` feels like it's related to key-value pairs but apparently this isn't doing that?

Reply:
I think there's definitely a argument for moving the Supabase stuff to one sensible place - I've been thinking about this recently. We should probably have a systematic approach though if we do this - for this PR I feel like they're better defined directly in ParcelsPage.

Reply:
Cool, think I agree

Reply:
have done